python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -691,7 +691,8 @@ def _flush(self, objs):
 
         for obj in objs:
             # check if the object is dirty, if not do nothing
-            if not obj._dirty:
+            this_id = self.find(obj)
+            if not obj._dirty or this_id not in self._loaded_ids:
```

Comment:
I'd personally prefer to check for the `obj._dirty` flag first to avoid a mildly expensive `find` operation on every object. Other than that, I'm happy with this I think given the reservations about `EmptyGangaObjects`